src/documents/FrameworkConcepts.md

```diff
@@ -30,7 +30,7 @@ It is possible to replace one object tree with another, but still to maintain a
 This is particularly important in web applications where data claiming to be "your data" can suddenly arrive from anywhere (a JSON feed, some persistence,
 a particularly aggressive version management system, etc). However it got here, you know it is your data because it is at the right path.
 
-EL paths within Infusion can be evaluated (dereferenced with respect to a particular model) by the framework utilities [`fluid.get()`](CoreAPI.md#fluid-get-model-path-) and [`fluid.set()`](CoreAPI.md#fluid-set-model-path-newvalue-),
+EL paths within Infusion can be evaluated (dereferenced with respect to a particular model) by the framework utilities [`fluid.get()`](CoreAPI.md#fluidgetmodel-path) and [`fluid.set()`](CoreAPI.md#fluidsetmodel-path-newvalue),
```

Comment:
I do like the new style of not requiring trailing "-" junk, but I can't quite yet spot the regularity behind the rule of which characters get converted to "-" and which don't

Reply:
The plugin we use ([`markdown-it-anchor`]()) uses the library [`string.js`](https://github.com/jprichardson/string.js), which provides a `slugify` function:

https://github.com/jprichardson/string.js/blob/master/lib/string.js#L417

That function calls a `dasherize` function from the same package:

https://github.com/jprichardson/string.js/blob/master/lib/string.js#L128

As I suspected from the variations I'd seen, it replaces one *or more* whitespace characters with a single dash.  Their algorithm seems to match what GitHub does in the 150 or so cases I tested, with the exception of [the one pattern I found](https://issues.fluidproject.org/browse/FLUID-6207).  We could point a link checker at GitHub to confirm if there are other variations, but unless we can find details about the logic GitHub itself uses (I couldn't in 15-20 minutes of searching), we would be making just as much of a guess ourselves in picking a replacement or writing our own plugin.

On a side note, in searching I discovered that GitLab (which a few of us have used for its CI integration) [helpfully publishes their rules](https://docs.gitlab.com/ee/user/markdown.html#header-ids-and-links), but insanely, even visually reviewing them it's obvious they don't match GitHub at all.

Reply:
I read through the GfM spec in some depth, the closest I could find is in [the section on links](https://github.github.com/gfm/#links):

> One label matches another just in case their normalized forms are equal. To normalize a label, strip off the opening and closing brackets, perform the Unicode case fold, strip leading and trailing whitespace and collapse consecutive internal whitespace to a single space. If there are multiple matching reference link definitions, the one that comes first in the document is used. (It is desirable in such cases to emit a warning.)

This sounds very close to what `string.js` does, with the additional step of replacing the remaining spaces with dashes.  However, obviously there is something in `fluid.log([logLevel, arg1, ... argn])` that is handled differently by the two, that causes `, ... ` to end up as a single dash with `string.js` and as a double-dash on GitHub (see [FLUID-6207](https://issues.fluidproject.org/browse/FLUID-6207)).